File: solutions/java/MissingNumber.java
public class MissingNumber {
    public static int missingNumber(int[] nums) {
        int n = nums.length;
        int expected = n*(n+1)/2;
        int sum = 0;
        for (int x : nums) sum += x;
        return expected - sum;
    }
    public static void main(String[] args) {
        System.out.println(missingNumber(new int[]{3,0,1}));
    }
}

File: solutions/cpp/missing_number.cpp
#include <iostream>
#include <vector>
using namespace std;

int missingNumber(vector<int>& nums) {
    int n = nums.size();
    int exp = n*(n+1)/2;
    int sum = 0;
    for(int x:nums) sum+=x;
    return exp - sum;
}

int main() {
    vector<int> v={3,0,1};
    cout<<missingNumber(v);
}
